gcc 9.2.1 ARM release build fails

Running:
```
$ /usr/bin/arm-linux-gnueabihf-gcc -v
Using built-in specs.
COLLECT_GCC=/usr/bin/arm-linux-gnueabihf-gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc-cross/arm-linux-gnueabihf/9/lto-wrapper
Target: arm-linux-gnueabihf
Configured with: ../src/configure -v --with-pkgversion='Debian 9.2.1-8' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c,ada,c++,go,d,fortran,objc,obj-c++,gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-libitm --disable-libquadmath --disable-libquadmath-support --enable-plugin --enable-default-pie --with-system-zlib --without-target-system-zlib --enable-libpth-m2 --enable-multiarch --disable-sjlj-exceptions --with-arch=armv7-a --with-fpu=vfpv3-d16 --with-float=hard --with-mode=thumb --disable-werror --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=arm-linux-gnueabihf --program-prefix=arm-linux-gnueabihf- --includedir=/usr/arm-linux-gnueabihf/include
Thread model: posix
gcc version 9.2.1 20190909 (Debian 9.2.1-8) 
```

I hit this building DR as part of drmemory:
```
drmemory-arm-rel-32: **** 9 build errors ****
	/.../src/dynamorio/core/arch/arm/encode.c:3063:9: error: ‘memset’ offset [73, 152] from the object at ‘di’ is out of the bounds of referenced subobject ‘errmsg’ with type ‘const char *’ at offset 68 [-Werror=array-bounds]
	make[2]: *** [dynamorio/core/CMakeFiles/drdecode.dir/build.make:167: dynamorio/core/CMakeFiles/drdecode.dir/arch/arm/encode.c.o] Error 1
	make[1]: *** [CMakeFiles/Makefile2:655: dynamorio/core/CMakeFiles/drdecode.dir/all] Error 2
	/.../src/dynamorio/core/arch/arm/encode.c:3063:9: error: ‘memset’ offset [73, 152] from the object at ‘di’ is out of the bounds of referenced subobject ‘errmsg’ with type ‘const char *’ at offset 68 [-Werror=array-bounds]
	make[2]: *** [dynamorio/core/CMakeFiles/dynamorio_static.dir/build.make:650: dynamorio/core/CMakeFiles/dynamorio_static.dir/arch/arm/encode.c.o] Error 1
	make[1]: *** [CMakeFiles/Makefile2:509: dynamorio/core/CMakeFiles/dynamorio_static.dir/all] Error 2
	/.../src/dynamorio/core/arch/arm/encode.c:3063:9: error: ‘memset’ offset [73, 152] from the object at ‘di’ is out of the bounds of referenced subobject ‘errmsg’ with type ‘const char *’ at offset 68 [-Werror=array-bounds]
	make[2]: *** [dynamorio/core/CMakeFiles/dynamorio.dir/build.make:650: dynamorio/core/CMakeFiles/dynamorio.dir/arch/arm/encode.c.o] Error 1
	make[1]: *** [CMakeFiles/Makefile2:580: dynamorio/core/CMakeFiles/dynamorio.dir/all] Error 2
```

--- a/core/arch/arm/encode.c
+++ b/core/arch/arm/encode.c
@@ -3060,7 +3060,9 @@ instr_encode_arch(dcontext_t *dcontext, instr_t *instr, byte *copy_pc, byte *fin
             return NULL;
         }
         /* We need to clear all the checking fields for each new template */
-        memset(&di.errmsg, 0, sizeof(di) - offsetof(decode_info_t, errmsg));
+        /* We avoid using "&di.errmsg" to avoid a gcc 9 warning (i#4170). */
+        memset(((char *)&di) + offsetof(decode_info_t, errmsg), 0,
+               sizeof(di) - offsetof(decode_info_t, errmsg));
     }
 
     /* Encode into di.instr_word */

